pkgs/by-name/aw/aw-watcher-window-wayland/package.nix

```diff
@@ -4,40 +4,51 @@
   rustPlatform,
   pkg-config,
   openssl,
+  stdenv,
+  darwin,
+  wayland,
+  unstableGitUpdater,
 }:
 rustPlatform.buildRustPackage rec {
   pname = "aw-watcher-window-wayland";
-  version = "6108ad3df8e157965a43566fa35cdaf144b1c51b";
+  version = "0-unstable-2024-10-08";
 
   src = fetchFromGitHub {
     owner = "ActivityWatch";
-    repo = pname;
-    rev = version;
-    hash = "sha256-xl9+k6xJp5/t1QPOYfnBLyYprhhrzjzByDKkT3dtVVQ=";
+    repo = "aw-watcher-window-wayland";
+    rev = "58bf86a6984cb01fa750c84ce468c7ccb167f796";
+    hash = "sha256-SnlShM44jnQiZGg5mjreZg1bsjFLNYMjC/krR1TXTE4=";
   };
 
-  cargoPatches = [ ./rustc-serialize-fix.patch ];
-
-  cargoLock = {
-    lockFile = ./Cargo.lock;
-    outputHashes = {
-      "aw-client-rust-0.1.0" = "sha256-9tlVesnBeTlazKE2UAq6dzivjo42DT7p7XMuWXHHlnU=";
-    };
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit pname version src;
+    hash = "sha256-WWT8tOrHPf5x3bXsVPt32VKut4qK+K8gickBfEc0zmk=";
   };
+
+  passthru.updateScript = unstableGitUpdater { };
+
   nativeBuildInputs = [
     pkg-config
   ];
 
-  buildInputs = [
-    openssl
-  ];
+  buildInputs =
+    [
+      openssl
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.Security
```

Comment:
Can we move to the new apple SDK

Reply:
We probably could, but this package is linux only, so I don't really see a need. 
It probably can't hurt either, but the package is not intended to run on macos.

Reply:
I don't know, it has the `platforms = lib.platforms.linux;`

Reply:
@LordGrimmauld perhaps wants to comment.

Reply:
True, we should probably drop the darwin things; I wrote this in the assumption it might work on darwin, but i do not have a way to test.
The upstream dev has since confirmed on discord this is *not intended for darwin* so i will remove this part.